diff -ur qpid-cpp-1.36.0/src/CMakeLists.txt qpid-cpp-1.36.0-edited/src/CMakeLists.txt
--- qpid-cpp-1.36.0/src/CMakeLists.txt	2016-12-09 17:16:02.000000000 +0000
+++ qpid-cpp-1.36.0-edited/src/CMakeLists.txt	2017-01-17 17:25:56.314054425 +0000
@@ -371,7 +371,7 @@
       qpid/sys/cyrus/CyrusSecurityLayer.h
       qpid/sys/cyrus/CyrusSecurityLayer.cpp
      )
-  include_directories (${CYRUS_SASL_INCLUDE_DIR})
+  include_directories (SYSTEM ${CYRUS_SASL_INCLUDE_DIR})
   set(sasl_LIB ${CYRUS_SASL_LIBRARY})
   set(HAVE_SASL 1)
 else (BUILD_SASL)
@@ -451,7 +451,7 @@
   # Add include directories and link directories for NSS
   # unfortunately this doesn't get done automatically for
   # libraries detected by FindPkgConfig
-  include_directories(${ssl_INCLUDES})
+  include_directories(SYSTEM ${ssl_INCLUDES})
   link_directories(${ssl_LIBDIRS})
 endif (BUILD_SSL)
 
